pymc3_hmm/utils.py

```diff
@@ -458,3 +457,59 @@ def plot_predictive_histograms(
             )
 
     return axes
+
+
+def plot_split_timeseries_hists(
+    plot_data: pd.DataFrame,
+    sample_col: str,
+    sample_mean_col: str,
+    twin_column_name: Union[str, None] = None,
+    observed_col: Union[str, None] = None,
+    **split_ts_kwargs
+):  # pragma: no cover
+    from matplotlib.ticker import MaxNLocator
+
+    def plot_fn(ax, data, **kwargs):
+        if observed_col is not None:
+            ax.plot(
+                data[observed_col],
+                label=observed_col,
+                alpha=0.7,
+                color="red",
+                linewidth=0.8,
+                drawstyle="steps",
+            )
+            # make sample mean less visible
+            alpha, color = 0.3, "blue"
+        else:
+            # make sample mean more visible
+            alpha, color = 0.7, "darkorange"
+
+        ax.plot(
+            data[sample_mean_col],
+            label=sample_mean_col,
+            alpha=alpha,
+            color=color,
+            linewidth=0.8,
+            drawstyle="steps",
+        )
```

Comment:
We need to allow the same sort of flexibility/customization as the functions we're calling (i.e. `plot_predictive_histograms` and `plot_split_timeseries`), which means that this function should also have a `plot_fn` argument that is called here.

This addition would allow users to provide a `plot_fn` that does what this code is currently doing, and much more.

Reply:
Looked into this. I guess it makes sense that we're adding a wrapper to the two functions we currently have. However, if we move this part (plus the `twin_ax` part at the bottom) out of this function, then it would be nothing more than `plot_split_timeseries` alone. If that's what we want, then should we just rename `plot_predictive_histograms` to something like `plot_timeseries_histograms`?

Reply:
Exactly.  We want this to be a very simple function that reduces the amount of boilerplate copy-and-paste required to plot sample-based time series.  

We need to do this in a way that doesn't reduce user's ability to customize something in a very specific way; otherwise, they won't be able to use this function and they'll need to rewrite parts of this function from scratch.

For instance, if a user doesn't agree with the color choices we've made here, then they can't use this helper function.  Likewise, if they need to add something to each of the split series, they won't be able to, because this helper function doesn't allow them to set their work `plot_fn`-like argument.

Remember, this function is just supposed to be a simple combination of `plot_split_timeseries` and `plot_predictive_histograms`.